src/librustc_data_structures/small_vec.rs

```diff
@@ -130,6 +130,18 @@ impl<A: Array> SmallVec<A> {
             self.set_len(len + 1);
         }
     }
+
+    pub fn truncate(&mut self, len: usize) {
+        unsafe {
+            while len < self.len() {
+                // Decrement len before the drop_in_place(), so a panic on Drop
+                // doesn't re-drop the just-failed value.
+                let len = self.len() - 1;
```

Comment:
I can change it, but it's also worth noting that I copied this code (with minor modifications) from vec.rs.